src/main/java/org/myrobotlab/framework/MethodCache.java

```diff
@@ -529,26 +531,50 @@ public List<MethodEntry> getRemoteOrdinalMethods(Class<?> object, String methodN
     return methodIndex.remoteOrdinalIndex.get(ordinalKey);
   }
 
-  public Object[] getDecodedJsonParameters(Class<?> clazz, String methodName, Object[] encodedParams) {
+  /**
+   * Decode parameters from a String Json format into the format
+   * specified by the declared method parameter type.
+   *
+   * <p>
+   *     If clazz or encodedParameters are null, then null is returned.
+   *     If methodName is null then NPE is thrown.
+   * </p>
+   *
+   * <p>
+   *     FIXME Change encodedParameters to String[] to enforce type safety
+   * </p>
+   *
+   * @param clazz The class to lookup methods for
+   * @param methodName The name of the method to decode parameters for
+   * @param encodedParams The encoded parameters in JSON format.
+   * @return The decoded parameters according to a matched method, or null
+   *  if no such method could be found.
+   */

```

Comment:
what is the info returned in the error when methodName parameter is null ? does the exception contain the npe parameter name ?

Reply:
Hmm, it appears the documentation there is mistaken, no NPE is thrown, instead only null is returned and an error logged. In that case should I keep the `@Nonnull` and fix the documentation or change it to nullable?

Reply:
:+1:

Reply:
Nonnull removed and documentation updated